crates/miden-tx/tests/integration/scripts/compatibility.rs

```diff
@@ -0,0 +1,490 @@
+
+    // Basic fungible faucet
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&p2id_note));
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&p2idr_note));
+    assert_eq!(CheckResult::No, faucet_account_interface.can_consume(&swap_note));
+
+    // STATEFUL CHECK
+    // --------------------------------------------------------------------------------------------
+
+    // TODO: implement
+}
+
+// CUSTOM NOTES
+// ================================================================================================
+
+#[test]
+fn test_basic_wallet_custom_notes() {
+    // STATELESS CHECK
+    // --------------------------------------------------------------------------------------------
+
+    let mut mock_chain = MockChain::new();
+    let wallet_account =
+        mock_chain.add_existing_wallet(Auth::BasicAuth, vec![FungibleAsset::mock(20)]);
+    let wallet_account_interface = AccountInterface::from(&wallet_account);
+
+    let sender_account_id =
+        ACCOUNT_ID_REGULAR_ACCOUNT_IMMUTABLE_CODE_ON_CHAIN_2.try_into().unwrap();
+    let serial_num =
+        RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]).draw_word();
+    let tag = NoteTag::from_account_id(wallet_account.id(), NoteExecutionMode::Local).unwrap();
+    let metadata = NoteMetadata::new(
+        sender_account_id,
+        NoteType::Public,
+        tag,
+        NoteExecutionHint::always(),
+        Default::default(),
+    )
+    .unwrap();
+    let vault = NoteAssets::new(vec![FungibleAsset::mock(100)]).unwrap();
+
+    let compatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.miden::contracts::faucets::basic_fungible->fungible_faucet
+
+        begin
+            push.1
+            if.true 
+                call.fungible_faucet::distribute
+                call.fungible_faucet::burn
+            else
+                call.wallet::receive_asset
+                call.wallet::create_note
+                call.wallet::move_asset_to_note
+            end
+        end
+    ";
+    let note_script =
+        NoteScript::compile(compatible_source_code, TransactionKernel::testing_assembler())
+            .unwrap();
+    let recipient = NoteRecipient::new(serial_num, note_script, NoteInputs::default());
+    let compatible_custom_note = Note::new(vault.clone(), metadata, recipient);
+    assert_eq!(
+        CheckResult::Maybe,
+        wallet_account_interface.can_consume(&compatible_custom_note)
+    );
+
+    let incompatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.miden::contracts::faucets::basic_fungible->fungible_faucet
+
+        begin
+            push.1
+            if.true 
+                call.fungible_faucet::distribute
+                call.fungible_faucet::burn
+            else
+                call.fungible_faucet::distribute
+                call.wallet::create_note
+                call.wallet::move_asset_to_note
```

Comment:
Can we add `call.wallet::receive_asset` so that the entire basic wallet interface is present? Just to make sure that the check result is `no` because of the presence of `distribute` and not because of the absence of `receive_asset`.